instruqt/protecting-your-assets/03-proxying-private-images/assignment.md

```diff
@@ -0,0 +1,251 @@
+
+![Managing Images on the Vendor Portal](../assets/managing-images.png)
+
+You use the "+ Add external registry" button to add a private registry. We won't
+be doing this in the lab environment since it's shared team.
+
+Adding the Proxy to Your Chart
+==============================
+
+The next step to securing your images with the Replicated Platform is to use
+the proxy to pull images. The Slackernews chart refers to its two images in
+it's default values. We'll change those to refer to the Replicated proxy
+registry.
+
+Images tags in the proxy registry follow the following pattern:
+
+```
+$PROXY_REGISTRY_HOST/proxy/$APP_SLUG/$ORIGINAL_IMAGE_TAG
+```
+
+so you'll proxy the image `ghcr.io/slackernews/slackernews-web:1.0.17` as
+
+```
+proxy.replicated.com/proxy/[[ Instruqt-Var key="REPLICATED_APP" hostname="shell" ]]/ghcr.io/slackernews/slackernews-web:1.0.17
+```
+
+for your instance of Slackernews. Let's do that now.
+
+### Updating Image References
+
+Select the "Slackernews Chart" tab to edit the chart source files. Select the
+file `values.yaml` and search for the image values.
+
+![Image Value Settings in Helm Chart](../assets/image-values.png)
+
+Update the image values to use the proxied image for the main Slackernews
+image
+
+```
+slackernews:
+  repository: proxy.replicated.com/proxy/[[ Instruqt-Var key="REPLICATED_APP" hostname="shell" ]]/ghcr.io/slackernews/slackernews-web:1.0.17
+```
+
+Since we've made a change that will impact our users, let's also bump the
+version number by editing `Chart.yaml` and setting a new chart version.
+
+```
+version: 0.5.0
+```
+
+The editor should have automatically saved your changes to both files.
+
+Allowing Image Access
+---------------------
+
+The last change you need to make is to assure an appropriate image pull secret
+is set on the pods that require access to proxied images. We need two things to
+do this:
+
+1. A template in our Helm chart to create the secret
+2. Reference to the secret for any workloads that need it
+
+Another good practice is to specify image pull secrets as Helm values, so we'll
+plan on doing that as well.
+
+The image pull secret provides registry credentials to pull image. An obvious
+question to ask is how we know which credentials to use for the proxy registry.
+Replicated takes care of this for is as part of the same process it uses to
+inject the license file into your Helm chart. This process provides a global
+value `replicated.dockerconfigjson` that contains the credentials.
+
+
+We're going to go through the changes above backwards, since Slackernews has
+already taken care of the second one. Open the `slackernews-deployment.yaml`
+file under templates and look for the following lines that use the image pull
+secret.
+
+```
+      {{ if hasKey ((.Values.global).replicated) "dockerconfigjson" }}
+      imagePullSecrets:
+        - name: replicated-pull-secret
+```
+
+This lines are only included if the global values is set, in other
```

Comment:
```suggestion
These lines are only included if the value `global.repliacted` is set, in other
```

Reply:
I'd like to keep this descriptive but it wasn't worded correctly. I'll update the way I'm thinking.